app/services/foreman_ansible/fact_importer.rb

```diff
@@ -55,10 +55,10 @@ def add_compose_fact(imported_values, fact_name, fact_fqn)
     end
 
     def missing_facts
-      db_fact_names = if db_facts.is_a? Hash
-                        db_facts.keys
+      db_fact_names = if host.fact_values.is_a? Hash
```

Comment:
it isn't a hash anymore so this condition and the following comment can be removed